cairo-contracts/packages/core/src/commitment/types.cairo

```diff
@@ -133,25 +133,19 @@ pub impl StateProofZero of Zero<StateProof> {
 
 #[derive(Clone, Debug, Drop, PartialEq, Serde, starknet::Store)]
 pub struct StateRoot {
-    pub root: ByteArray // TODO: Determine the correct type (ByteArray or Array<u8>) once implemented membership proof verification.
-}
-
-pub impl ByteArrayIntoRoot of Into<ByteArray, StateRoot> {
-    fn into(self: ByteArray) -> StateRoot {
-        StateRoot { root: self }
-    }
+    pub root: [u32; 8],
```

Comment:
add a comment why `[u32; 8]` was preferred over `Array<u8>`.

note that in general, we may have a state root bigger than 256 bytes.